services/publicapi/service.go

```diff
@@ -65,6 +65,9 @@ func (s *service) SendTransaction(input *services.SendTransactionInput) (*servic
 	s.reporting.Info("transaction received via public api", log.String("flow", "checkpoint"), log.Stringable("txHash", txHash))
 	defer s.reporting.Info("transaction status returned by public api", log.String("flow", "checkpoint"), log.Stringable("txHash", txHash))
 
+	meter := s.reporting.Meter("tx-processing-time", log.Stringable("txHash", txHash))
+	defer meter.Done()
```

Comment:
I'm not sure this is good enough. Do we want to collect different metrics, or to qualify at least, the difference between a tx that has finished successfully, a tx that timed out (i.e. result did not return within timeout interval) and a tx that failed? if so, how would we do that?

Reply:
This is a good point

Reply:
we can move forward with this, but document the downsides fo this approach in a new issue